openedx/event-routing-backends
# Allow event-routing-backends to be used when transforming non-openedx events

**Description:** 

Event Routing Backends is designed to be installed with edx-platform, and the code assumes this. But we want to use ERB outside of the platform too -- it contains event transform classes and test logic which is useful when transforming non-openedx events.

This PR:
* Catches ImportErrors from edx-platform to fix errors like [these](https://github.com/open-craft/openedx-completion-aggregator/actions/runs/9462018403/job/26064058779?pr=205)
* Adds a utility function to add custom events to the backend whitelists.
* Refactors transformer test mixin classes to make them available for use outside of ERB.

See https://github.com/open-craft/openedx-completion-aggregator/pull/205 for an example usage.

**JIRA:** [FAL-3766](https://tasks.opencraft.com/browse/FAL-3766) (private-ref)

Part of https://github.com/openedx/openedx-aspects/issues/222

**Merge deadline:** end of June

**Testing instructions:**

See https://github.com/open-craft/openedx-completion-aggregator/pull/205 for manual test instructions.

**Merge checklist:**

**Post merge:**
      finished.
